components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java

```diff
@@ -517,6 +517,58 @@ public IdentityProvider getIdPByRealmId(String realmId, int tenantId,
         return identityProvider;
     }
 
+    /**
+     * Get the enabled IDP of the given realm id.
+     *
+     * @param realmId       Realm ID of the required identity provider.
+     * @param tenantId      Tenant ID of the required identity provider.
+     * @param tenantDomain  Tenant domain of the required identity provider.
+     * @return              Enabled identity provider of the given realm id.
+     * @throws IdentityProviderManagementException Error when getting the identity provider.
+     */
+    public IdentityProvider getEnabledIdPByRealmId(String realmId, int tenantId,
+                                            String tenantDomain) throws IdentityProviderManagementException {
+
+        IdPHomeRealmIdCacheKey cacheKey = new IdPHomeRealmIdCacheKey(realmId);
+        IdPCacheEntry entry = idPCacheByHRI.getValueFromCache(cacheKey, tenantDomain);
+        if (entry != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Cache entry found for Identity Provider with Home Realm ID " + realmId);
+            }
+            // Check whether the idp in the cache is enabled.
+            if (entry.getIdentityProvider().isEnable()) {
+                return entry.getIdentityProvider();
+            }
+            if (log.isDebugEnabled()) {
```

Comment:
no need to do the `isDebugEnabled()` check, if the string operations in the message are not costly.

Reply:
Check this in other places as well

Reply:
I added the isDebugEnabled() check since I was told that the concat operations are considered as costly. Is it not needed even when there are concat operations?